qemu/tests/blockdev_snapshot_data_file.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from virttest import error_context
+from virttest import data_dir
+
+from provider import backup_utils
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+from provider.virt_storage.storage_admin import sp_admin
+
+
+class BlkSnapshotWithDatafile(BlockDevSnapshotTest):
+
+    def __init__(self, test, params, env):
+        super(BlkSnapshotWithDatafile, self).__init__(test, params, env)
+        self.trash = []
+
+    def prepare_snapshot_file(self):
+        params = self.params.copy()
+        params.setdefault("target_path", data_dir.get_data_dir())
+        image_with_datafile = backup_utils.create_image_with_data_file
+        images = image_with_datafile(self.main_vm, params, self.snapshot_tag)
+        self.trash.extend(images)
+
+    def post_test(self):
+        try:
+            self.clone_vm.destroy()
+            for image in self.trash:
+                sp_admin.remove_volume(image)
+        except Exception as error:
+            logging.error(str(error))
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Block device snapshot test
+    1) Start VM with a data disk
+    2) Create snapshot target with data file
+    3) Do snapshot and save a temp file and record md5sum
```

Comment:
Suggest to add a step to check data-file exists by 'query-block' since this is the test point.

Reply:
Have update, @ngu-niny, please help to review again.